command-parser/src/parser.rs

```diff
@@ -90,16 +90,44 @@ impl<'a> Parser<'a> {
     ///
     /// [`Command`]: struct.Command.html
     pub fn parse(&'a self, buf: &'a str) -> Option<Command<'a>> {
-        let (prefix, padding) = self.find_prefix(buf)?;
-        let mut idx = prefix.len();
-
-        match buf.get(idx..)? {
-            v if !v.starts_with(padding) => return None,
-            _ => {}
-        }
-
-        idx += padding.len();
+        let (prefix, _) = self.find_prefix(buf)?;
+        self.parse_with_prefix(prefix, buf)
+    }
 
+    /// Parse a command out of a buffer with a specific prefix.
+    ///
+    /// Instead of using the list of set prefixes, give a specific prefix
+    /// to parse the message, this can be used to have a kind of dynamic
+    /// prefixes.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// # use twilight_command_parser::{Command, CommandParserConfig, Parser};
+    /// # fn example() -> Option<()> {
+    /// let mut config = CommandParserConfig::new();
+    /// config.add_prefix("!");
+    /// config.add_command("echo", false);
+    ///
+    /// let parser = Parser::new(config);
+    ///
+    /// let command = parser.parse_with_prefix("=", "=echo foo")?;
+    ///
+    /// assert_eq!("=", command.prefix);
+    /// assert_eq!("echo", command.name);
+    /// # Some(())
+    /// # }
+    /// ```
+    ///
+    /// [`Command`]: struct.Command.html
+    pub fn parse_with_prefix(
+        &'a self,
+        prefix: &'a str,
+        buf: &'a str,
+    ) -> Option<Command<'a>> {
+        if !buf.starts_with(prefix) { return None }
```

Comment:
I can't load the CI logs to see what's causing the rustfmt error but this is probably it